Add OpInfos for torch.{complex, view_as_real, view_as_complex}

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56524 Add OpInfos for torch.{complex, view_as_real, view_as_complex}**

Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

anjali411 committed:
Add OpInfos for torch.{complex, view_as_real, view_as_complex}

[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"




[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"




[ghstack-poisoned]

anjali411 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -5674,14 +5698,8 @@ def unpack_variables(args):
 
 
 EXCLUDE_FUNCTIONAL = {
-    'addmm',
```
cleanup

anjali411 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -2493,13 +2503,13 @@ def sample_inputs_matmul(op_info, device, dtype, requires_grad):
 
     samples = (
-        SampleInput(_make_tensor_helper((S, S), low=0), args=(_make_tensor_helper((S, S)),)),
-        SampleInput(_make_tensor_helper((), low=0), args=(_make_tensor_helper(()),)),
```
TODO: undo this change since abs (first input to polar) should be non-negative

mruberry commented on test/test_autograd.py:
```diff
@@ -5368,7 +5368,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
 # and only run for floating point
 
-separate_complex_tests = ['view_as_real', 'div', '__rdiv__', 'sub']
+separate_complex_tests = ['div', '__rdiv__', 'sub']
```
Wow; we're almost done with this list! That's cool

mruberry reviewed (commented):
Awesome!

Some failures are real, however. 

Lint: 

```
./torch/testing/_internal/common_methods_invocations.py:3388:17: E126 continuation line over-indented for hanging indent
```

the new polar grad and gradgrad tests are failing, and a skip is needed for:

test_supported_backward_view_as_complex_cuda_float16

The skip and lint fix should be easy. For polar grad and gradgrad this could revert modifying its sample inputs and possible file an issue?

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"




[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"




[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"


Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"


Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

[ghstack-poisoned]

mruberry reviewed (approved):
Awesome!

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"


Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

[ghstack-poisoned]

anjali411 committed:
Update on "Add OpInfos for torch.{complex, view_as_real, view_as_complex}"


Differential Revision: [D27909165](https://our.internmc.facebook.com/intern/diff/D27909165)

[ghstack-poisoned]
